src/app/map/map.component.ts

```diff
@@ -47,109 +46,54 @@ export class MapComponent implements OnInit, OnChanges {
       attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
     }).addTo(this.map);
 
-    this.updateMarkers();
-    this.initRouting();
   }
 
 
   private updateMarkers(): void {
     if (this.map) {
-      // This clears existing markers
+      // Clear existing markers
       this.map.eachLayer((layer) => {
         if (layer instanceof L.Marker) {
           this.map.removeLayer(layer);
         }
       });
-
-      // This adds new markers
-      this.markers.forEach(coords => {
-        L.marker(coords).addTo(this.map)
-          .bindPopup('Location ' + coords)
-          .openPopup();
-
-        // EXPERIMENTAL 2 
-      // this.markers.forEach(marker => {
-      //   L.marker(marker.coords).addTo(this.map)
-      //     .bindPopup(marker.name)
-      //     .openPopup();
-      });
+      
+      const cordinateset = this.markers.slice(4, 14).map(coords => [coords[0], -coords[1]]);
+      this.initRouting(cordinateset)
     }
   }
 
-
-  private initRouting(): void {
-    if (this.markers.length > 1) {
-       // This initializes routing control to navigate from one point to another 
-       const routingOptions: L.Routing.RoutingControlOptions = {
-
-        waypoints: this.markers.map(coords => L.latLng(coords[0], coords[1])), // Another point in Disney World
-        routeWhileDragging: true,
-
+  private initRouting(cordinateset: any[][]): void {
+        this.routingControl = L.Routing.control({
         router: L.Routing.osrmv1({
           language: 'en', // This sets the language for OSRM router 
           profile: 'foot-walking' // This sets a custom walking profile
         }),
-        plan: L.Routing.plan(this.markers.map(coords => L.latLng(coords[0], coords[1])), {
-          createMarker: (i: number, waypoint: L.Routing.Waypoint) => {
-            return L.marker(waypoint.latLng, {
-             draggable: true
-            }).bindPopup('Waypoint ' + (i + 1));
-          },
-          language: 'en' // This sets the language for plan 
+         collapsible: true,
+         plan: L.Routing.plan([
```

Comment:
Here are the markers. I limit the amount of waypoints and then splice at the end. Once again, I'm not sure if this was the issue but the map renders much better.